### PR TITLE
Hide user+password in serialized URLs in UI

### LINKED ISSUE #1403: Do not show the `user:pass@` from bitcoind node in the setup UI
This is a subset of https://github.com/fedimint/fedimint/issues/1298 but important and urgent

### PR DESCRIPTION
**Pull request by elsirion**
Fixes #1403

### CODE REVIEW

File: fedimint-core/src/util.rs [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+    pub fn new_borrowed(url: &'a Url) -> SanitizedUrl<'a> {
+        SanitizedUrl(Cow::Borrowed(url))
+    }
+}
+
+impl<'a> Display for SanitizedUrl<'a> {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}://", self.0.scheme())?;
+
```
Comment (dpc) [inline fedimint-core/src/util.rs:27]: Might be better to print `REDACTED` or something when user or password are set.

**Review by dpc — APPROVED** [review]

File: fedimint-core/src/util.rs [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        let test_cases = vec![
+            ("http://1.2.3.4:80/foo", "http://1.2.3.4/foo", "SanitizedUrl(http://1.2.3.4/foo, has_password=false, has_username=false)"),
+            ("http://1.2.3.4:81/foo", "http://1.2.3.4:81/foo", "SanitizedUrl(http://1.2.3.4:81/foo, has_password=false, has_username=false)"),
+            ("fedimint://1.2.3.4:1000/foo", "fedimint://1.2.3.4:1000/foo", "SanitizedUrl(fedimint://1.2.3.4:1000/foo, has_password=false, has_username=false)"),
+            ("fedimint://foo:bar@domain.com:1000/foo", "fedimint://domain.com:1000/foo", "SanitizedUrl(fedimint://domain.com:1000/foo, has_password=true, has_username=true)"),
+            ("fedimint://foo@1.2.3.4:1000/foo", "fedimint://1.2.3.4:1000/foo", "SanitizedUrl(fedimint://1.2.3.4:1000/foo, has_password=false, has_username=true)"),
+        ];
+
+        for (url_str, sanitized_display_expected, sanitized_debug_expected) in test_cases {
+            dbg!(&url_str);
```
Comment (elsirion) [inline fedimint-core/src/util.rs:70]: Please complain about such `dbg!` statements, that was left here by accident.

Comment (dpc) [reply fedimint-core/src/util.rs:70]: I haven't noticed, but if it's in tests, then I guess it's fine? :D


Maybe we should have a pre-commit check looking for `dbg!`. I leave them accidentally all the time. :D

Comment (elsirion) [reply fedimint-core/src/util.rs:70]: I ran the tests after the merge and it just looked ugly :laughing: thx for adding the check!

**Pull request merged**